### PR TITLE
Upgrade: `10up-toolkit` to `v4`

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change

* This PR upgrades the `10up-toolkit` to version 4.
* Runs `npm update`.
* Also adds a few dev deps which are recommended while building assets in the Github actions.
* Also fixing https://github.com/10up/publisher-media-kit/security/dependabot/28

### Alternate Designs

n/a

### Possible Drawbacks

n/a

### Verification Process

Plugin functionalities should work without fail.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Security - Upgrade `10up-toolkit` to `v4`

### Credits

Props @jeffpaul @faisal-alvi

### CODE REVIEW

**Review by Sidsector9 — CHANGES_REQUESTED** [review]
@faisal-alvi there is a conflict it seems. Can you resolve?

File: package.json [hunk 1]
```diff
@@ -18,19 +18,31 @@
...
+    "eslint": "^8.1.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-config-prettier": "^6.15.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsdoc": "^37.0.3",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-prettier": "^3.2.0",
+    "eslint-plugin-react": "^7.30.0",
+    "eslint-plugin-react-hooks": "^2.5.0",
```
Comment (Sidsector9) [inline package.json]: @faisal-alvi a lot of these seem to be dev dependencies, should we move them?

**Comment by faisal-alvi** [comment]
A new dependabot alert related to `semver-regex` is found here: https://github.com/10up/publisher-media-kit/security/dependabot/30. This PR will hopefully fix this issue. cc: @Sidsector9

**Review by Sidsector9 — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (faisal-alvi) [reply package.json]: @Sidsector9 removed it (https://github.com/10up/publisher-media-kit/pull/85/commits/dc446b725b6f0f69b52f54eb2eac9764b63d7794) and now the Github action is failing again, it requires these dependencies in order to lint properly. I've added them in https://github.com/10up/publisher-media-kit/pull/85/commits/3fd6908c285f96dc2ebdb0bb33b7fc1a353b8b9f which had fixed the issue. Any suggestions?

Comment (Sidsector9) [reply package.json]: @faisal-alvi  What I meant is, dev dependencies were added under dependencies. We can just move them under dev dependencies.

Comment (faisal-alvi) [reply package.json]: Oh! I read `should we remove them?` instead of `should we move them?` 😄

**Comment by faisal-alvi** [comment]
Over to you @Sidsector9

**Review by Sidsector9 — APPROVED** [review]

**Pull request merged**